lib/portage/process.py

```diff
@@ -706,10 +706,10 @@ def _exec(binary, mycommand, opt_name, fd_pipes,
 						if unshare_net:
 							# use 'localhost' to avoid hostname resolution problems
 							try:
-								# pypy3 does not implement socket.sethostname()
+								# Old pypy3 versions do not implement socket.sethostname()
 								new_hostname = b'localhost'
 								if hasattr(socket, 'sethostname'):
-									socket.sethostname(new_hostname)
+									socket.sethostname(new_hostname.decode("ascii"))
```

Comment:
Let's use `portage._native_string(new_hostname)`, since that's our exiting convention for anything that requires a native string.

Reply:
I think we should wait for @mgorny. Apparently his PyPy patch was wrong and PyPy should take bytes as an argument too.